### PR TITLE
chore: refine renovate config

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

Using `renovate` locally I was able to make sure all `1926` versions we have in our monorepo are either tracked by renovate, or generated, for a total of of `373` tracked dependencies in `82` files !

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -134,28 +140,44 @@
...
   "ignorePaths": [
     "**/algoliasearch/**",
     "**/client-abtesting/**",
     "**/client-analytics/**",
     "**/client-insights/**",
+    "**/client_insights/**",
     "**/client-personalization/**",
     "**/client-query-suggestions/**",
     "**/client-search/**",
+    "**/client_search/**",
```
Comment (shortcuts) [inline renovate.json:180]: how come they are duplicated?

File: renovate.json [hunk 2]
```diff
@@ -134,28 +140,44 @@
...
       "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
     }
   ],
+  "github-actions": {
+    "fileMatch": [
+      "\\.github/workflows/.*\\.ya?ml$"
+    ]
+  },
   "ignorePaths": [
     "**/algoliasearch/**",
```
Comment (shortcuts) [inline renovate.json:172]: does it allow negative patterns? maybe we can ignore everything in packages but the common ones

**Review by shortcuts — COMMENTED** [review]
THIS IS HUGE

File: renovate.json (hunk 2 above)
Comment (millotp) [reply renovate.json:172]: Unfortunately no it's only `glob`, not even regex, and the `includePaths` property doesn't work alongside it so we can't do like in `generation.config.mjs`

File: renovate.json (hunk 1 above)
Comment (shortcuts) [reply renovate.json:180]: Ahhhhh

**Review by shortcuts — APPROVED** [review]
Gg!!

**Pull request merged**